data-products/src/main/scala/org/sunbird/analytics/job/report/CollectionSummaryJob.scala

```diff
@@ -0,0 +1,195 @@
+  }
+  // $COVERAGE-ON$
+
+  def getUserData(spark: SparkSession, fetchData: (SparkSession, Map[String, String], String, StructType) => DataFrame): DataFrame = {
+    val schema = Encoders.product[UserData].schema
+    fetchData(spark, userCacheDBSettings, "org.apache.spark.sql.redis", schema)
+      .withColumn("username", concat_ws(" ", col("firstname"), col("lastname"))).persist(StorageLevel.MEMORY_ONLY)
+  }
+
+  def getUserEnrollment(spark: SparkSession, fetchData: (SparkSession, Map[String, String], String, StructType) => DataFrame): DataFrame = {
+    fetchData(spark, userEnrolmentDBSettings, cassandraUrl, new StructType())
+      .select(col("batchid"), col("userid"), col("courseid"), col("active")
+        , col("completionpercentage"), col("enrolleddate"), col("completedon"), col("status"), col("certificates"), col("issued_certificates")).persist(StorageLevel.MEMORY_ONLY)
+  }
+
+  def prepareReport(spark: SparkSession, fetchData: (SparkSession, Map[String, String], String, StructType) => DataFrame, batchList: List[String])(implicit fc: FrameworkContext, config: JobConfig): DataFrame = {
+    implicit val sparkSession: SparkSession = spark
+    implicit val sqlContext: SQLContext = spark.sqlContext
+    import spark.implicits._
+    val userCachedDF = getUserData(spark, fetchData = fetchData).select("userid", "userchannel").persist(StorageLevel.MEMORY_ONLY)
+    val processBatches: DataFrame = filterBatches(spark, fetchData, config, batchList)
+      .join(getUserEnrollment(spark, fetchData), Seq("batchid", "courseid"), "left_outer")
+      .join(userCachedDF, Seq("userid"), "left_outer").drop("completionpercentage", "active")
+
+    val courseIds = processBatches.select(col("courseid")).distinct().collect().map(_ (0)).toList.asInstanceOf[List[String]]
+    val courseInfo = CourseUtils.getCourseInfo(courseIds, 500).toDF("framework", "identifier", "name", "channel", "batches", "organisation")
+    JobLogger.log(s"Total courseInfo records ${courseInfo.count()}", None, INFO)
+    val filteredBatches = processBatches.join(courseInfo, processBatches.col("courseid") === courseInfo.col("identifier"), "inner")
+      .select(processBatches.col("*"), courseInfo.col("identifier"), courseInfo.col("channel"), courseInfo.col("name"), courseInfo.col("organisation"))
+      .withColumn("batchid", concat(lit("batch-"), col("batchid")))
+      .withColumn("collectionName", col("name"))
+      .withColumn("publishedBy", concat_ws(", ", col("organisation")))
+      .withColumn("isPDFCertificatedIssued", when(col("certificates").isNotNull && size(col("certificates").cast("array<map<string, string>>")) > 0, true).otherwise(false))
+      .withColumn("isSVGCertificatedIssued", when(col("issued_certificates").isNotNull && size(col("issued_certificates").cast("array<map<string, string>>")) > 0, true).otherwise(false))
```

Comment:
Refer to the progressexhaust job logic and convert them into isCertificateIssued flag directly